text/0000-temp-components-lib.md

```diff
@@ -0,0 +1,618 @@
+4. Team leads that haven't been able to keep up with the technology trends but still need to code when possible.
+5. Teams that have the experience and understanding of Javascript and React, but don't have the time that may be required to use an API like Headless-styles (vs. dropping in an abstraction of it - component).
+
+### What is the expected outcome?
+
+The expected outcome of creating a component library should be:
+
+1. More adoption for Pando due to having a library that uses a familar pattern that people are generally more comfortable with.
+2. Should naturally teach users about the semantic token naming convention as it will inheritently use Headless-styles for prop definitions (which uses the same naming convention of the themes for
+consistency).
+3. Provide a developer experience that uses "React best practices" for a result that naturally allows scalability with custom designs via composable components, and high performance via advanced React patterns.
+4. Share the functionality as Lego blocks (create your own outcome based on what you combine/remove).
+5. By nature, it should increase the community contribution for the same reasons
+mentioned above (a familiar design).
+
+## Detailed design
+
+The design of the components library should first and foremost center around the
+Pando standards:
+
+1. A11y above all
+2. High Perfomance
+3. Easy to use
+
+Outside of these standards, there are 2 groups of components to consider:
+
+1. [Simple components](#simple-components) (Tag, Button, TextLink, etc.)
+2. [Complex compents](#complex-components) (Alert Dialogs, Notifications, Modal, etc.)
+
+Last, there should also be helpers that naturally compliment the experience:
+
+1. [Custom hooks](#custom-hooks)
+2. [Helper functions](#helper-functions)
+
+_:bulb: In addition to React 18, all components should be natively compatible for use
+within React Server Components._
+
+### Simple Components
+
+Simple components should mimick the vanilla JSX design React provides with the Headless-styles
+props extending it (i.e. not manipulating React props or creating "component
+does all" solutions).
+
+For example, a Button should be built something like this:
+
+```typescript
+import { getButtonProps } from '@pluralsight/headless-styles'
+import type { ButtonOptions } from '@pluralsight/headless-styles/types'
+import { type HTMLAttributes, type PropsWithChildren } from 'react'
+
+interface ButtonProps extends ButtonOptions, HTMLAttributes<HTMLButtonElement> {}
+
+function PandoButton(props: PropsWithChildren<ButtonProps>, ref: RefObject) {
+  const {
+    className,
+    disabled,
+    icon,
+    sentiment,
+    size,
+    usage,
+    ...nativeBtnProps
+  } = props
+  const pandoProps = getButtonProps({
+    className,
+    disabled,
+    sentiment,
+    size,
+    usage,
+    icon
+  })
+
+  return (
+    <button {...pandoProps.button} {...nativeProps} ref={ref} />
```

Comment:
Minor typo? Should this be `nativeBtnProps` (or at least the same as the destructured prop above)?